File: app/src/main/java/com/parabola/newtone/mvp/view/QueueView.java
package com.parabola.newtone.mvp.view;

import com.parabola.domain.model.Track;
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView;

import java.util.List;

import moxy.MvpView;
import moxy.viewstate.strategy.alias.AddToEndSingle;
import moxy.viewstate.strategy.alias.OneExecution;

@AddToEndSingle
public interface QueueView extends MvpView {

    void refreshTracks(List<Track> tracks);
    void setItemViewSettings(TrackItemView viewSettings);
    void setItemDividerShowing(boolean showed);

    void setTrackCount(int tracksCount);
    void setCurrentTrackPosition(int currentTrackPosition);

    @OneExecution
    void goToItem(int itemPosition);
}

File: app/src/main/java/com/parabola/newtone/mvp/view/QueueView.kt
package com.parabola.newtone.mvp.view

import com.parabola.domain.model.Track
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView
import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle
import moxy.viewstate.strategy.alias.OneExecution

@AddToEndSingle
interface QueueView : MvpView {
    fun setCurrentTrackPosition(currentTrackPosition: Int)
    fun refreshTracks(tracks: List<Track>)
    fun setTrackCount(tracksCount: Int)

    fun setItemViewSettings(viewSettings: TrackItemView)
    fun setItemDividerShowing(showed: Boolean)

    @OneExecution
    fun goToItem(itemPosition: Int)
}
